services/thermal_mgr/thermal_mgr.c

```diff
@@ -21,51 +21,100 @@ static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
 static QueueHandle_t thermalMgrQueueHandle;
 static StaticQueue_t thermalMgrQueueBuffer;
 static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
-
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
   /* Send an event to the thermal manager queue */
 
+  if (thermalMgrQueueHandle == NULL)
+    return ERR_CODE_UNKNOWN;
+
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *)event, (TickType_t)0) == errQUEUE_FULL)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
-  /* Implement this function */
+void osHandlerLM75BD(void)
+{
+
+  float temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
```

Comment:
In an ISR, you shouldn't call any function that could take too long or call an RTOS function with a non-zero block time. Since `osHandlerLM75BD` is assumed to execute from within an ISR, there are 2 problems here:
1. I2C transfers are pretty slow. If an ISR's execution time is too long, it's more likely that you'll delay other interrupts.
2. `readTempLM75BD` calls `i2cSendTo` and `i2cReceiveFrom`. Both attempt to take a mutex with a non-zero block time. 

So you shouldn't call `readTempLM75BD` from `osHandlerLM75BD`